go/vt/vtctl/reparentutil/emergency_reparenter.go

```diff
@@ -420,9 +421,11 @@ func (erp *EmergencyReparenter) findMostAdvanced(
 	winningPosition := tabletPositions[0]
 
 	// We have already removed the tablets with errant GTIDs before calling this function. At this point our winning position must be a
-	// superset of all the other valid positions. If that is not the case, then we have a split brain scenario, and we should cancel the ERS
+	// superset of all the other valid positions - unless the tablet with that position is taking a backup, in which case it might
+	// have a larger GTID set but we are still not choosing it. If that is not the case, then we have a split brain scenario, and
+	// we should cancel the ERS
 	for i, position := range tabletPositions {
-		if !winningPosition.AtLeast(position) {
+		if !winningPosition.AtLeast(position) && !backingUpTablets[topoproto.TabletAliasString(validTablets[i].Alias)] {
```

Comment:
Idiomatic go would be to do
```
_, ok := backingUpTablets[....]
```
and use `!ok` in the if condition.

Reply:
This map has all tablets, irrespective of whether they are or are not taking a backup. We need to check the value stored in the map for the alias of the current tablet, so the `if` needs to check what's inside `backingUpTablets[topoproto.TabletAliasString(validTablets[i].Alias)]`. I can add a safety check to ensure that the element is. in the map, though it should always be given how it's created.

Reply:
Fixed